### ISSUE TITLE
[IMPROVEMENT] - Restoring position without a delay

### ISSUE THREAD

**Issue by Yoksven** [Sanqui/obsidian-persistent-graph#5]
Currently position restores after a seconds of graph doing it's thing. It would be better if it would happen right away, before any graph animation could kick in. I don't know if there's any API limitation that prevents it, so if it's not possible, I understand, but otherwise, I'd like to request an improvement.

**Comment by Sanqui** [comment]
The technical reason for the delay is the fact that Obsidian takes a while to load all nodes onto the graph, and if I restore positions too early, the nodes added later won't register their new position.  Obsidian provides me with no API that would tell me whether all the nodes have loaded or not.  So I wait a bit and use some heuristics to determine whether any new nodes were added in the last 600ms.  I could potentially shorten this time, but I would need to test it on a vault with a lot of nodes, and/or a slow computer.  Another thing I could attempt to do is to restore positions multiple times even while the graph is loading, but I suspect it might be just as jumpy...

**Comment by Yoksven** [comment]
Yeah. I suspected that something was in the way. Shortening the time for some milliseconds won't change much. I'm somewhat curious to see "the jumpy way" of loading over and over. If there isn't much of a performance penalty, it might be better, but emphasis on "might".
